include/algorithms/elastic_net/elastic_net_training_batch.h

```diff
@@ -0,0 +1,190 @@
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  Implementation of the interface for elastic net model-based training in the batch processing mode
+//--
+*/
+
+#ifndef __ELASTIC_NET_TRAINING_BATCH_H__
+#define __ELASTIC_NET_TRAINING_BATCH_H__
+
+#include "algorithms/algorithm.h"
+#include "services/daal_defines.h"
+#include "services/daal_memory.h"
+#include "algorithms/elastic_net/elastic_net_training_types.h"
+#include "algorithms/elastic_net/elastic_net_model.h"
+#include "algorithms/linear_model/linear_model_training_batch.h"
+
+namespace daal
+{
+namespace algorithms
+{
+namespace elastic_net
+{
+namespace training
+{
+namespace interface1
+{
+/**
+ * @defgroup elastic_net_batch Batch
+ * @ingroup elastic_net_training
+ * @{
+ */
+/**
+ * <a name="DAAL-CLASS-ALGORITHMS__ELASTIC_NET__TRAINING__BATCHCONTAINER"></a>
+ * \brief Class containing methods for normal equations elastic net model-based training using algorithmFPType precision arithmetic
+ */
+template <typename algorithmFPType, Method method, CpuType cpu>
+class BatchContainer : public TrainingContainerIface<batch>
+{
+public:
+    /**
+     * Constructs a container for elastic net model-based training with a specified environment in the batch processing mode
+     * \param[in] daalEnv   Environment object
+     */
+    BatchContainer(daal::services::Environment::env * daalEnv);
+
+    /** Default destructor */
+    ~BatchContainer();
+
+    /**
+     * Computes the result of elastic net model-based training in the batch processing mode
+     *
+     * \return Status of computations
+     */
+    services::Status compute() DAAL_C11_OVERRIDE;
+};
+
+/**
+ * <a name="DAAL-CLASS-ALGORITHMS__ELASTIC_NET__TRAINING__BATCH"></a>
+ * \brief Provides methods for elastic net model-based training in the batch processing mode
+ * <!-- \n<a href="DAAL-REF-ELASTICNET-ALGORITHM">Elastic net algorithm description and usage models</a> -->
+ *
+ * \tparam algorithmFPType  Data type to use in intermediate computations for elastic net model-based training, double or float
+ * \tparam method           Elastic net training method, \ref Method
+ *
+ * \par Enumerations
+ *      - \ref Method  Computation methods
+ *
+ * \par References
+ *      - \ref elastic_net::interface1::Model "elastic_net::Model" class
+ *      - \ref prediction::interface1::Batch "prediction::Batch" class
+ */
+template <typename algorithmFPType = DAAL_ALGORITHM_FP_TYPE, Method method = defaultDense>
+class DAAL_EXPORT Batch : public linear_model::training::Batch
```

Comment:
operator= ?

Reply:
In this class you have a ptr and delete inside dtor;
in case of accident copy of obj you can got a double free

Reply:
it was fixed